theano/sandbox/gpuarray/elemwise.py

```diff
@@ -256,19 +232,15 @@ def c_code(self, node, name, inputs, outputs, sub):
             broadcasts = map(int, node.inputs[idx].broadcastable)
             broadcasts = ', '.join(map(str, broadcasts))
             nd = node.inputs[idx].ndim
-            if nd > 0:
-                code += """
-                int broadcasts_%(iname)s[%(nd)s] = {%(broadcasts)s};
-                """ % locals()
-            else:
-                code += """
-                int *broadcasts_%(iname)s = NULL;
-                """ % locals()
+            code += """
+            int broadcasts_%(iname)s[%(nd)s+1] = {%(broadcasts)s};
+            """ % locals()
             emitted_inames[iname] = node.inputs[idx]
 
         # check that all inputs have valid dimensions
         emitted_inames = {}
         for idx, iname in enumerate(inputs):
+            code += "rargs[%(idx)s] = &%(iname)s->ga;\n" % dict(idx=idx, iname=iname)
```

Comment:
Then we should add something here to not reuse inputs that are the same, probably.  This would cut down on indexing overhead.  But it can be part of future code.